internal/controller/biosversionset_controller.go

```diff
@@ -229,6 +230,42 @@ func (r *BIOSVersionSetReconciler) deleteOrphanBIOSVersions(
 	return warnings, errors.Join(errs...)
 }
 
+func (r *BIOSVersionSetReconciler) patchOrCreateBIOSVersionfromTemplate(
+	ctx context.Context,
+	log logr.Logger,
+	biosVersionTemplate *metalv1alpha1.BIOSVersionTemplate,
+	biosVersionList *metalv1alpha1.BIOSVersionList,
+) error {
+	if len(biosVersionList.Items) == 0 {
+		log.V(1).Info("No BIOSVersion found, skipping spec template update")
+		return nil
+	}
+
+	var errs []error
+	for _, biosVersion := range biosVersionList.Items {
+		if biosVersion.Status.State != metalv1alpha1.BIOSVersionStateInProgress {
```

Comment:
since there is no `else`, we could check for in progress and use continue?
```
if biosVersion.Status.State == metalv1alpha1.BIOSVersionStateInProgress {
  continue
}
```